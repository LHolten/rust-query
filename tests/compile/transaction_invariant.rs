use rust_query::{Database, LocalClient, Table, migration::schema};

#[schema]
enum Schema {
    MyTable { name: String },
}
use v0::*;

fn test(db: Database<Schema>) {
    let mut client = LocalClient::try_new().unwrap();

    let txn = client.transaction(&db);
    let name = txn.query(|rows| {
        let item = MyTable::join(rows);

        txn.query_one(item.name())
    });

    println!("{name}");
}

fn main() {}
