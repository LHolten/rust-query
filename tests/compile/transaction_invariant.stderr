error[E0521]: borrowed data escapes outside of closure
  --> tests/compile/transaction_invariant.rs:17:9
   |
13 |     let txn = client.transaction(&db);
   |         --- `txn` declared here, outside of the closure body
14 |     let name = txn.query(|rows| {
   |                           ---- `rows` is a reference that is only valid in the closure body
...
17 |         txn.query_one(item.name())
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^ `rows` escapes the closure body here
